src/App.js

```diff
@@ -49,33 +52,40 @@ class ItemsListAndFilterMenu extends React.Component {
 
 
 class FilterMenu extends React.Component {
-    onInputCheckedJeans = () => {
-        this.props.jFilter()
+    onFilterChange = () => {
+        this.props.filter()
     }
     render() {
+        let filter = this.onFilterChange
+        const TYPES = [
+            {
+                type: 'Джинсы',
+            },
+            {
+                type: 'Рубашки',
+            },
+            {
+                type: 'Костюмы',
+            },
+        ]
+        let goodsTemplate = TYPES.map(function(item) {
+            return(
+                <React.Fragment>
+                    <li>
+                        <label htmlFor="jeans">
+                            <input type="checkbox" id="jeans" onChange={filter}/>
+                            <span>{item.type}</span>
```

Comment:
`id=jeans` looks like an error. You should use type instead of static value. Also onChange should pass item type to callback.